src/tribler-core/tribler_core/components/bandwidth_accounting/bandwidth_accounting_component.py

```diff
@@ -17,36 +13,34 @@
 class BandwidthAccountingComponent(RestfulComponent):
     community: BandwidthAccountingCommunity
 
-    _ipv8: IPv8
+    _ipv8_component: Ipv8Component
 
     async def run(self):
         await super().run()
         await self.get_component(UpgradeComponent)
-        config = self.session.config
-
-        ipv8_component = await self.require_component(Ipv8Component)
-        self._ipv8 = ipv8_component.ipv8
-        peer = ipv8_component.peer
+        self._ipv8_component = await self.require_component(Ipv8Component)
 
+        config = self.session.config
         if config.general.testnet or config.bandwidth_accounting.testnet:
             bandwidth_cls = BandwidthAccountingTestnetCommunity
         else:
             bandwidth_cls = BandwidthAccountingCommunity
 
         db_name = "bandwidth_gui_test.db" if config.gui_test_mode else f"{bandwidth_cls.DB_NAME}.db"
         database_path = config.state_dir / STATEDIR_DB_DIR / db_name
-        database = BandwidthDatabase(database_path, peer.public_key.key_to_bin())
-        community = bandwidth_cls(peer, self._ipv8.endpoint, self._ipv8.network,
-                                  settings=config.bandwidth_accounting,
-                                  database=database)
-        self._ipv8.add_strategy(community, RandomWalk(community), 20)
+        database = BandwidthDatabase(database_path, self._ipv8_component.peer.public_key.key_to_bin())
+        self.community = bandwidth_cls(self._ipv8_component.peer,
+                                       self._ipv8_component.ipv8.endpoint,
+                                       self._ipv8_component.ipv8.network,
```

Comment:
**thought/nit:** We're injecting `self._ipv8_component.ipv8.network` as a dependency and then the component itself is overwriting its input with a `Network()` of its own creation. We should, at some point (probably not this PR) move the `Network()` overwrite to this file (same for `PopularityCommunity`).

Reply:
Yes, probably. For me purpose of overwriting a Network is unclear.
It is true for `PopularityCommunity` and `GigachannelCommunity` (as successors of `RemoteQueryCommunity`)
cc: @ichorid 

(I see it as out of the scope of this PR)

Reply:
GigaChannelCommunity walks and deletes peers aggressively, so creates a separate Network instance to prevent interfering with other communities.

Reply:
The reason for a "clean" `Network()` is the introduction of bias through custom parameterization of Communities. If you bias your peer discovery mechanism, you get a self-reinforcing effect that essentially forces some peers out of your network. It is not a problem if one peer deviates, but if you force a default on the entire network you will start seeing this. We've learned this the hard way https://github.com/Tribler/tribler/issues/2894#issuecomment-462808735 Therefore, for every Community that uses custom walks (including peer removal) or custom parameterization, we use a new `Network`.

Reply:
Implemented here: https://github.com/Tribler/tribler/pull/6408